documents/entity/handler.go

```diff
@@ -156,6 +156,41 @@ func (h *grpcHandler) Get(ctx context.Context, getRequest *cliententitypb.GetReq
 	return resp, nil
 }
 
+func (h *grpcHandler) GetEntityByRelationship(ctx context.Context, getRequest *cliententitypb.GetRequestRelationship) (*cliententitypb.EntityResponse, error) {
+	apiLog.Debugf("Get request %v", getRequest)
+	ctxHeader, err := contextutil.Context(ctx, h.config)
+	if err != nil {
+		apiLog.Error(err)
+		return nil, err
+	}
+
+	entityIdentifier, err := hexutil.Decode(getRequest.Identifier)
+	if err != nil {
+		apiLog.Error(err)
+		return nil, centerrors.Wrap(err, "identifier is an invalid hex string")
+	}
+
+	erIdentifier, err := hexutil.Decode(getRequest.RelationshipIdentifier)
+	if err != nil {
+		apiLog.Error(err)
+		return nil, centerrors.Wrap(err, "identifier is an invalid hex string")
+	}
+
+	model, err := h.service.RequestEntityWithRelationship(ctxHeader, entityIdentifier, erIdentifier)
```

Comment:
I still need to add the following check:
Only request via p2p if the entity is not in the database or if the stored entity is not the latest version.